### DISCUSSION TITLE
Help to find out board name and available FW

### DISCUSSION THREAD

**Discussion by troyane** [roleoroleo/yi-hack-Allwinner-v2#653] [General]
Hello! I've got a few of these Noname _(as far as I can see relatively new)_ cameras. 

![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/477468/6fb4278c-2b6e-429e-a3bd-a7abe36cd4e4)

Trying to figure out if it is possible to update its FW to something better. 

Thanks!

**Comment by roleoroleo** [comment]
It's not possible to find the model from board name.
Probably it's a b091qp but I need a dump of the firmware.
Please, make a backup and share with me.

**Reply by troyane** [reply]
Thanks for the quick answer! Here it is: 
[mtd.zip](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/11846737/mtd.zip)

**Reply by troyane** [reply]
To be honest, I thought that markings on the very bottom `B102QP...` defines the model.

**Reply by roleoroleo** [reply]
The firmware matches the cam described here:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/281#issuecomment-1019278053
suffix model: b091qp
fw: 9.0.41.99_202103301400

It should be supported.

**Reply by troyane** [reply]
Thanks a lot! I'll try to update FW and will write you back if all was successful.

**Reply by troyane** [reply]
I updated FW. **I'm really glad that it works!** 

I understand that it's another topic, but my **Synology NAS can't pick it up over ONVIF.** Can't see it via automatic search either can't auth via manual setup. Seen that it's not a new topic [here](https://github.com/roleoroleo/yi-hack-MStar/issues/55). Were that changes already incorporated into [b091qp_0.2.7.tgz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/releases/download/0.2.7/b091qp_0.2.7.tgz)?

**Reply by roleoroleo** [reply]
I completely rewritten onvif software in the last release.
I could try to improve the code but I need help.
Are you able to capture the network traffic between Synology and cam? For example with tcpdump.

EDIT

If you need tcpdump for the cam, try here:
https://github.com/roleoroleo/yi-hack-utils/tree/main/Allwinner-v2
Download, decompress and run it with:
`tcpdump -i wlan0 -w /tmp/output.pcap`

**Reply by troyane** [reply]
Sure, @roleoroleo I can. I'll provide you with this information. Just don't want to paste it here. Can we talk in private?

**Reply by roleoroleo** [reply]
Ok, write me with Telegram.
But I'm not online very often this week.

**Reply by roleoroleo** [reply]
Please try this beta:
[onvif.tar.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/11972765/onvif.tar.gz)

**Reply by roleoroleo** [reply]
@troyane Any news about onvif?

**Reply by troyane** [reply]
Hello, @roleoroleo. Can you give more instructions, please? Should I put it on the SD card only?

**Reply by roleoroleo** [reply]
Yes, you should overwrite current files inside the SD card.
In the folder yi-hack

**Reply by troyane** [reply]
@roleoroleo it works! Synology was able to find ONVIF device in the network by itself. I was able to add a camera. Tested it, and PTZ also works great!

I have another camera of the same type, going to test it too, and will write you back.

**Reply by roleoroleo** [reply]
Thank you for your feedback.
It the patch is confirmed I will release a new version asap.

**Reply by troyane** [reply]
I don't know if it is related, but camera is rebooting each ~5mins. What kind of logs will be helpful?

**Reply by roleoroleo** [reply]
Is swap enabled?

**Reply by troyane** [reply]
Yes, swap file is enabled.

**Reply by troyane** [reply]
@roleoroleo I Just did another round of tests. Here is the timing:
- 20:50 turned on
- 21:40 reboot
- 2146 reboot
- 21:51 reboot
- 21:56 reboot
- ... and so on

**Reply by roleoroleo** [reply]
Probably something related to snapshots.
Did you enable snapshot? Timelapse?
Or a client that call a snapshot through web services?

**Reply by troyane** [reply]
Yes, snapshots were turned on. No timelapses. I've turned off snapshots. 

Checking how does it work now...

**Reply by troyane** [reply]
It behaves exactly the same as described here https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/653#discussioncomment-6419715

**Reply by roleoroleo** [reply]
Please, try to run these commands.
I would like to check the running processes:
```
ps
df
top
```

**Reply by troyane** [reply]
Sure, @roleoroleo  here you are:
 
- [df.txt.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/12049183/df.txt.gz)
- [ps.txt.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/12049185/ps.txt.gz)
- [top.log.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/12049186/top.log.gz)

`top` command run via: `top -b -d 1 -n 20 > top.log 2>&1`

**Reply by roleoroleo** [reply]
Logs are ok.
I don't know how to debug this behavior.
Maybe running some top when the cam is close to restarting.

**Reply by troyane** [reply]
In that 5 min in-between time, I've logged via SSH to cam and got this message:
`mount: mounting none on /sys/kernel/debug failed: Resource busy` 


Also, here are the commands you've asked for, executed just before reboot:
- [df.txt.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/12053362/df.txt.gz)
- [top.log.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/12053363/top.log.gz)
- [ps.txt.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/12053364/ps.txt.gz)

**Reply by roleoroleo** [reply]
ps output shows the problem.
rmm process is dead and wd_rtsp.sh (the watchdog) starts the reboot process.
So, the problem is: why does rmm die?
Normally it happens when there is a memory problem (not enough free memory).

Please, try to switch off all unnecessary processes.

**Reply by troyane** [reply]
From the very beginning I switched off mqtt, ftp,... Left the web service and ssh turned on. 
Here are few thoughts:
- It happens almost as according to a schedule: 1h cam works, then each 5min it reboots.
- I'll try to turn off webservice and test again. 
- Is there an option to use dmesg? Maybe it could have some more detail on a problem?
- If you have already updated the firmware I can try to set it up from scratch.

**Comment by roleoroleo** [comment]
>     * Is there an option to use dmesg? Maybe it could have some more detail on a problem?

Yes, copy the busybox binbary:
```
cp /tmp/sd/yi-hack/bin/busybox /tmp/sd/yi-hack/bin/dmesg
dmesg
```

>     * If you have already updated the firmware I can try to set it up from scratch.

Try new 0.2.8 released 1 hour ago.

I checked your log files again and the problem is not simple.
Normally (with my cams) rmm process uses about 65% of VSZ.
Your log shows 86%, it's strange.
AFAIK b091qp cams are a customization made by other vendors (not directly by Yi).
Maybe rmm is not so stable.
We should try to know why it works without the hack.

**Reply by troyane** [reply]
I tried new fw, and everything worked from the very beginning. Thanks for your efforts!
